File: Beecrowd_Parte4/Perolas_1975.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

#define MAX_ALUNOS 100
#define MAX_PEARLS 1000
#define MAX_NAME_LENGTH 101
#define MAX_ANSWER_LENGTH 1001

// Função para encontrar o índice de um aluno no vetor de nomes
int indexOf(char nomes[][MAX_NAME_LENGTH], int numAlunos, char nome[]) {
    for (int i = 0; i < numAlunos; i++) {
        if (strcmp(nomes[i], nome) == 0) {
            return i;
        }
    }
    return -1; // Aluno não encontrado
}

int main() {
    int P, A, R; // número de pérolas, número de alunos, número de respostas por aluno
    char pearls[MAX_PEARLS][MAX_ANSWER_LENGTH];
    char nomes[MAX_ALUNOS][MAX_NAME_LENGTH];
    int ocorrencias[MAX_ALUNOS] = {0};

    while (1) {
        scanf("%d %d %d", &P, &A, &R);
        if (P == 0 && A == 0 && R == 0) break;

        // Ler as pérolas escritas no quadro (não precisamos armazenar)
        for (int i = 0; i < P; i++) {
            scanf(" %[^\n]", pearls[i]);
        }

        // Ler os nomes dos alunos
        for (int i = 0; i < A; i++) {
            scanf(" %[^\n]", nomes[i]);

            // Ler as respostas dos alunos e contar ocorrências de "ERRO"
            for (int j = 0; j < R; j++) {
                char resposta[MAX_ANSWER_LENGTH];
                scanf(" %[^\n]", resposta);

                // Verificar se a resposta contém "ERRO"
                if (strstr(resposta, "ERRO") != NULL) {
                    ocorrencias[i]++;
                }
            }
        }

        // Encontrar o máximo de ocorrências
        int maxOcorrencias = -1;
        for (int i = 0; i < A; i++) {
            if (ocorrencias[i] > maxOcorrencias) {
                maxOcorrencias = ocorrencias[i];
            }
        }

        // Imprimir os alunos com mais ocorrências (em ordem alfabética)
        int primeiro = 1;
        for (int i = 0; i < A; i++) {
            if (ocorrencias[i] == maxOcorrencias) {
                if (!primeiro) {
                    printf(", ");
                }
                printf("%s", nomes[i]);
                primeiro = 0;
            }
        }
        printf("\n\n"); // linha em branco após cada caso de teste
    }
    return 0;
}

File: Beecrowd_Parte4/Perolas_1975.java
import java.util.*;

public class Perolas_1975 {

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        while (true) {
            int P = scanner.nextInt(); // número de pérolas
            int A = scanner.nextInt(); // número de alunos
            int R = scanner.nextInt(); // número de respostas por aluno

            if (P == 0 && A == 0 && R == 0) break;

            // Ler as pérolas
            scanner.nextLine(); // para consumir o newline após os ints
            Map<String, Integer> ocorrencias = new HashMap<>();

            for (int i = 0; i < P; i++) {
                scanner.nextLine(); // ignorar a pérola
            }

            // Ler os alunos e suas respostas
            for (int i = 0; i < A; i++) {
                String nomeAluno = scanner.nextLine().trim(); // ler nome do aluno
                int count = 0;

                for (int j = 0; j < R; j++) {
                    String resposta = scanner.nextLine();
                    if (resposta.contains("ERRO")) {
                        count++;
                    }
                }

                // Atualizar contagem de ocorrências do aluno
                ocorrencias.put(nomeAluno, count);
            }

            // Encontrar o aluno com mais aparições
            int maxCount = -1;
            List<String> melhoresAlunos = new ArrayList<>();

            for (Map.Entry<String, Integer> entry : ocorrencias.entrySet()) {
                int count = entry.getValue();
                if (count > maxCount) {
                    maxCount = count;
                    melhoresAlunos.clear();
                    melhoresAlunos.add(entry.getKey());
                } else if (count == maxCount) {
                    melhoresAlunos.add(entry.getKey());
                }
            }

            // Ordenar os alunos em ordem alfabética
            Collections.sort(melhoresAlunos);

            // Imprimir resultado
            for (int i = 0; i < melhoresAlunos.size(); i++) {
                if (i > 0) {
                    System.out.print(", ");
                }
                System.out.print(melhoresAlunos.get(i));
            }
            System.out.println(); // imprimir linha em branco após cada caso de teste
        }

        scanner.close();
    }
}
